src/interface/inference.py

```diff
@@ -0,0 +1,180 @@
+                        help="Running as REPL or process corpus."
+                             " [repl, corpus]")
+    parser.add_argument('--input_path', type=str, default=None,
+                        help="Input path for running inference."
+                             " One input per line."
+                             " Only in eval_mode='corpus'")
+    parser.add_argument('--output_path', type=str, default=None,
+                        help="Output path for running inference."
+                             " Only in eval_mode='corpus'")
+
+    return parser.parse_args(cl_arguments)
+
+
+def main(cl_arguments):
+    ''' Run REPL for a CoLA model '''
+    cl_args = handle_arguments(cl_arguments)
+    args = config.params_from_file(cl_args.config_file, cl_args.overrides)
+
+    # Raise error if obsolete arg names are present
+    check_arg_name(args)
+
+    if args.cuda >= 0:
+        try:
+            if not torch.cuda.is_available():
+                raise EnvironmentError("CUDA is not available, or not detected"
+                                       " by PyTorch.")
+            log.info("Using GPU %d", args.cuda)
+            torch.cuda.set_device(args.cuda)
+        except Exception:
+            log.warning(
+                "GPU access failed. You might be using a CPU-only"
+                " installation of PyTorch. Falling back to CPU."
+            )
+            args.cuda = -1
+
+    # Prepare data #
+    pretrain_tasks, target_tasks, vocab, word_embs = build_tasks(args)
+    tasks = sorted(set(pretrain_tasks + target_tasks), key=lambda x: x.name)
+
+    # Build or load model #
+    model = build_model(args, vocab, word_embs, tasks)
+
+    assert args.load_eval_checkpoint != "none"
+    log.info("Loading existing model from %s...", args.load_eval_checkpoint)
+    load_model_state(model, args.load_eval_checkpoint, args.cuda, [],
+                     strict=False)
+
+    # Inference Setup #
+    model.eval()
+    vocab = Vocabulary.from_files(os.path.join(args.exp_dir, 'vocab'))
+    indexers = build_indexers(args)
+    task = take_one(tasks)
+    assert task.name == "cola", \
+        "Currently only supporting CoLA. ({})".format(task.name)
+
+    if cl_args.inference_mode == "repl":
+        assert cl_args.input_path is None
+        assert cl_args.output_path is None
+        print("Running REPL for task: {}".format(task.name))
+        repl_loop(model, vocab, indexers, task, args)
+    elif cl_args.inference_mode == "corpus":
+        run_inference(model, vocab, indexers, task, args,
+                      cl_args.input_path, cl_args.output_path)
+    else:
+        raise KeyError(cl_args.inference_mode)
+
+
+def repl_loop(model, vocab, indexers, task, args):
+    print("Input CTRL-C or enter 'QUIT' to terminate.")
+    while True:
+        try:
+            input_string = input(" INPUT: ")
+            if input_string == "QUIT":
+                break
+
+            tokens = process_sentence(
```

Comment:
Not all tasks use this input processing. If we're sticking to that abstraction (processing done on the Task object), maybe refactor so this calls a `task.process_input(...)` method?

Reply:
Should I leave a raise NotImplementedError on the remaining tasks? I feel like it might take some work before I can implement a more general abstraction applicable across tasks.

Reply:
yeah, that's fine.